File: src/main/java/com/wafflestudio/draft/dto/response/UserInformationResponse.java
package com.wafflestudio.draft.dto.response;

import com.wafflestudio.draft.model.User;
import lombok.Data;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

@Data
@RequiredArgsConstructor
public class UserInformationResponse {
    @NonNull
    private Long id;
    @NonNull
    private String username;
    @NonNull
    private String email;
    private String profileImage;

    public UserInformationResponse(User user) {
        this.id = user.getId();
        this.username = user.getUsername();
        this.email = user.getEmail();
        this.profileImage = user.getProfileImage();
    }
}

File: src/main/kotlin/com/wafflestudio/draft/dto/response/UserInformationResponse.kt
package com.wafflestudio.draft.dto.response

import com.wafflestudio.draft.model.User

data class UserInformationResponse (
    var id: Long,
    var username: String,
    var email: String,
    var profileImage: String
){
    constructor(user:User){
        id= user.id!!
        username=user.username
        email=user.email
        profileImage=user.profileImage
    }
}